File: WWF/Hands on labs 3.5/Lab06/VB/after/Exercise2/ContosoWorkflows/processPOWorkflow.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" as from Windows Workflow Foundation
' Hands on Labs RC
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

Public Class processPOWorkflow
    Inherits SequentialWorkflowActivity
    Public newPO As POSchema.PO = POSchema.PO.GeneratePOInstance()
    Public outgoingPOforFabrikam As New ContosoWorkflows.Fabrikam.PO
    Public POResponsefromFabrikam As  New ContosoWorkflows.Fabrikam.PO

    Public outgoingPOforNorthwind As New ContosoWorkflows.Northwind.PO
    Public POResponsefromNorthwind As New ContosoWorkflows.Northwind.PO

    Private Sub invokePOSubmissionWSFabrikam_Invoked(ByVal sender As System.Object, ByVal e As System.Workflow.Activities.InvokeWebServiceEventArgs)

    End Sub

    Private Sub invokePOSubmissionWSFabrikam_Invoking(ByVal sender As System.Object, ByVal e As System.Workflow.Activities.InvokeWebServiceEventArgs)



        POHelpers.CopyPOHeader(newPO, outgoingPOforFabrikam)
        POHelpers.CopyPOHistory(newPO, outgoingPOforFabrikam)
        POHelpers.CopyPOItems(newPO, outgoingPOforFabrikam)



    End Sub


    
    Private Sub invokePOSubmissionWSNorthwind_Invoked(ByVal sender As System.Object, ByVal e As System.Workflow.Activities.InvokeWebServiceEventArgs)

    End Sub
    Private Sub invokePOSubmissionWSNorthwind_Invoking(ByVal sender As System.Object, ByVal e As System.Workflow.Activities.InvokeWebServiceEventArgs)




        POHelpers.CopyPOHeader(newPO, outgoingPOforNorthwind)
        POHelpers.CopyPOHistory(newPO, outgoingPOforNorthwind)
        POHelpers.CopyPOItems(newPO, outgoingPOforNorthwind)



    End Sub

    Private Sub afterWSFabrikamInvoke_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)



        'Initialize it to a time where electronic PO processing was not invented.
        Dim statusDate As New DateTime(1900, 1, 1)
        Dim lastStatus As Fabrikam.POStatus = Nothing

        'Find the last status by looping over the history items
        For Each stat As Fabrikam.POStatus In POResponsefromFabrikam.History

            If (stat.Timestamp > statusDate) Then

                statusDate = stat.Timestamp
                lastStatus = stat
            End If

        Next
        System.Console.WriteLine("Your PO has been {0} on {1} by Fabrikam", lastStatus.PoStatus, lastStatus.Timestamp)



    End Sub

    Private Sub afterWSNorthwindInvoke_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)



        'Initialize it to a time where electronic PO processing was not invented.
        Dim statusDate As New DateTime(1900, 1, 1)
        Dim lastStatus As Northwind.POStatus = Nothing

        'Find the last status by looping over the history items
        For Each stat As Northwind.POStatus In POResponsefromNorthwind.History

            If (stat.Timestamp > statusDate) Then

                statusDate = stat.Timestamp
                lastStatus = stat
            End If

        Next
        System.Console.WriteLine("Your PO has been {0} on {1} by Northwind", lastStatus.PoStatus, lastStatus.Timestamp)



    End Sub
End Class

File: WWF/Hands on labs 3.5/Lab06/CSharp/after/Exercise 1/ContosoWorkflows/processPOWorkflow.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" as from Windows Workflow Foundation
// Hands on Labs RC
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace ContosoWorkflows
{
	public sealed partial class processPOWorkflow: SequentialWorkflowActivity
	{
        public POSchema.PO newPO = POSchema.PO.GeneratePOInstance();

		public processPOWorkflow()
		{
			InitializeComponent();
		}

        public ContosoWorkflows.Fabrikam.PO outgoingPOforFabrikam = new ContosoWorkflows.Fabrikam.PO();
        public ContosoWorkflows.Fabrikam.PO POResponsefromFabrikam = new ContosoWorkflows.Fabrikam.PO();

        private void invokePOSubmissionWSFabrikam_Invoking(object sender, InvokeWebServiceEventArgs e)
        {
            POHelpers.CopyPOHeader(newPO, outgoingPOforFabrikam);
            POHelpers.CopyPOHistory(newPO, outgoingPOforFabrikam);
            POHelpers.CopyPOItems(newPO, outgoingPOforFabrikam);

        }

        private void invokePOSubmissionWSFabrikam_Invoked(object sender, InvokeWebServiceEventArgs e)
        {

        }

        public ContosoWorkflows.Northwind.PO outgoingPOforNorthwind = new ContosoWorkflows.Northwind.PO();
        public ContosoWorkflows.Northwind.PO POResponsefromNorthwind = new ContosoWorkflows.Northwind.PO();

        private void invokePOSubmissionWSNorthwind_Invoking(object sender, InvokeWebServiceEventArgs e)
        {
            outgoingPOforNorthwind = new Northwind.PO();
            POHelpers.CopyPOHeader(newPO, outgoingPOforNorthwind);
            POHelpers.CopyPOHistory(newPO, outgoingPOforNorthwind);
            POHelpers.CopyPOItems(newPO, outgoingPOforNorthwind);

        }

        private void invokePOSubmissionWSNorthwind_Invoked(object sender, InvokeWebServiceEventArgs e)
        {

        }

        private void afterWSFabrikamInvoke_ExecuteCode(object sender, EventArgs e)
        {
            // Initialize it to a time where electronic PO processing was not invented.
            DateTime statusDate = new DateTime(1900, 1, 1);
            Fabrikam.POStatus lastStatus = null;

            //Find the last status by looping over the history items
            foreach (Fabrikam.POStatus stat in POResponsefromFabrikam.History)
            {
                if (stat.Timestamp > statusDate)
                {
                    statusDate = stat.Timestamp;
                    lastStatus = stat;
                }
            }

            System.Console.WriteLine("Your PO has been {0} on {1} by Fabrikam", lastStatus.PoStatus, lastStatus.Timestamp);

        }

        private void afterWSNorthwindInvoke_ExecuteCode(object sender, EventArgs e)
        {
            // Initialize it to a time where electronic PO processing was not invented.
            DateTime statusDate = new DateTime(1900, 1, 1);
            Northwind.POStatus lastStatus = null;

            //Find the last status by looping over the history items
            foreach (Northwind.POStatus stat in POResponsefromNorthwind.History)
            {
                if (stat.Timestamp > statusDate)
                {
                    statusDate = stat.Timestamp;
                    lastStatus = stat;
                }
            }
            System.Console.WriteLine("Your PO has been {0} on {1} by Northwind", lastStatus.PoStatus, lastStatus.Timestamp);

        }

  
	}

}
